File: src/js/core/src/Utils/Xml.ts
/**
 * Minimal XML escaping helper for SVG/XML text and attribute content.
 */
export class Xml {
  static #entities: Record<string, string> = {
    '&': '&amp;',
    "'": '&apos;',
    '"': '&quot;',
    '<': '&lt;',
    '>': '&gt;',
  };

  static #pattern = new RegExp(
    `[${Object.keys(Xml.#entities).join('')}]`,
    'g',
  );

  /**
   * Returns `value` with the five XML predefined entities escaped.
   */
  static escape(value: string): string {
    return value.replace(Xml.#pattern, (ch) => Xml.#entities[ch]);
  }
}

File: src/dart/core/lib/src/utils/xml.dart
/// Minimal XML escaping helper for SVG/XML text and attribute content.
library;

const _entities = <String, String>{
  '&': '&amp;',
  "'": '&apos;',
  '"': '&quot;',
  '<': '&lt;',
  '>': '&gt;',
};

final _pattern = RegExp('[&\'"<>]');

/// Returns [value] with the five XML predefined entities escaped, in a
/// single pass like the JS port's regex replace.
String escapeXml(String value) {
  return value.replaceAllMapped(_pattern, (match) => _entities[match[0]]!);
}
